src/components/Search/SearchResultHooks.js

```diff
@@ -0,0 +1,614 @@
+    link,
+    logo: iconByContentType('sources', t, link),
+    content: renderSnippet(link, highlight, null /* No default description */, t),
+    views,
+    t,
+  };
+
+  return <SearchResultOneItem {...props} />;
+});
+
+export const SearchResultLandingPage = withNamespaces()(({ landingPage, filterValues, t }) => {
+  const link = landingPageSectionLink(landingPage, filterValues);
+
+  const linkTitle = SEARCH_GRAMMAR_LANDING_PAGES_SECTIONS_TEXT[landingPage] || 'home.sections';
+  const valuesTitleSuffix = (filterValues && filterValues.filter(filterValue => filterValue.name !== 'text').map(filterValue => filterValue.origin || filterValue.value).join(' ')) || '';
+  const subText = t(SEARCH_GRAMMAR_LANDING_PAGES_SECTIONS_SUBTEXT[landingPage]);
+
+  const props = {
+    key: landingPage,
+    title: `${t(linkTitle)} ${valuesTitleSuffix}`,
+    link,
+    logo: iconByContentType(SEARCH_GRAMMAR_LANDING_PAGES_SECTIONS_CONTENT_TYPE[landingPage], t, link),
+    content: renderSnippet(link, null /* No highlights for landing pages. */, subText, t),
+    t,
+  };
+
+  return <SearchResultOneItem {...props} />;
+});
+
+export const SearchResultOneItem = withNamespaces()(({ key, title, link, logo, content, part, parts, date, views, collectionTitle, collectionLink, t }) => {
+  const description = [];
+  collectionTitle && description.push(collectionTitle);
+  part && description.push(t('pages.unit.info.episode', { name: part }));
+  parts && description.push(`${parts} ${t('pages.collection.items.programs-collection')}`);
+  date && description.push(t('values.date', { date }));
+  !!views && views > 0 && description.push(t('pages.unit.info.views', { views }));
+
+  return (<List.Item key={key} className="media_item">
+    {logo}
+    <div className="media_item__content">
+      <TooltipIfNeed text={title} Component={Header} as={Link} to={link} content={title} />
+      {content && (<TooltipIfNeed text={content} Component={Container} content={content} />)}
+      <div className={clsx('description', { 'is_single': !(description?.length > 1) })}>
+        {description.map((d, i) => (<span key={i}>{d}</span>))}
+        {collectionLink && (<span className="opacity_1">
+          <Link as={'a'} to={collectionLink}>
+            {/* ARTICLES should have different text, then "To all episodes..." should be "To all articles..." */}
+            {t('programs.list.show_all')}
+          </Link>
+        </span>)}
+      </div>
+    </div>
+  </List.Item>);
+});
+
+const getFilterById = (getTagById, getSourceById) => index => {
+  switch (index) {
+    case SEARCH_INTENT_INDEX_TOPIC:
+      return getTagById;
+    case SEARCH_INTENT_INDEX_SOURCE:
+      return getSourceById;
+    default:
+      return x => x;
+  }
+};
+
+export const SearchResultIntent = withNamespaces()(({ id, name, type, index, highlight, t }) => {
+  const namespace = `intents_${id}_${type}`;
+  const dispatch = useDispatch();
+  useEffect(() => {
```

Comment:
If you don't add parameter to useEffect trigger its can take old one in closure 
[documentation](https://reactjs.org/docs/hooks-effect.html#:~:text=Experienced%20JavaScript%20developers,this%20page.)

I was make this mistake and plane fixe it